common/mam/mask.c

```diff
@@ -0,0 +1,32 @@
+#include <string.h>
+
+#include "common/trinary/add.h"
+#include "mam/mask.h"
+
+int mask(trit_t *dest, trit_t *message, size_t l, Curl *const c) {
+  size_t chunk_length;
+  trit_t chunk[HASH_LENGTH];
+  for (size_t i = 0; i < l; i += HASH_LENGTH) {
+    chunk_length = l - i < HASH_LENGTH ? l - i : HASH_LENGTH;
+    memcpy(chunk, &message[i], chunk_length * sizeof(trit_t));
+    for (size_t j = 0; j < chunk_length; j++) {
+      dest[i + j] = trit_sum(chunk[j], c->state[j]);
```

Comment:
can't you just use `message[i+j]` here and remove the local `chunk` buffer?

Reply:
It would remove the ability to mask in-place